crossterm_cursor/src/lib.rs

```diff
@@ -1,13 +1,449 @@
+    #[cfg(unix)]
+    cursor: AnsiCursor,
+}
+
+impl TerminalCursor {
+    /// Creates a new terminal cursor.
+    pub fn new() -> TerminalCursor {
+        #[cfg(windows)]
+        let cursor = if supports_ansi() {
+            Box::new(AnsiCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        } else {
+            Box::new(WindowsCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        };
+
+        #[cfg(unix)]
+        let cursor = AnsiCursor::new();
+
+        TerminalCursor { cursor }
+    }
+
+    /// Moves the cursor to the given position.
+    pub fn goto(&self, x: u16, y: u16) -> Result<()> {
+        self.cursor.goto(x, y)
+    }
+
+    /// Gets the cursor position.
+    pub fn pos(&self) -> Result<(u16, u16)> {
+        self.cursor.pos()
+    }
+
+    /// Moves the cursor `row_count` times up.
+    pub fn move_up(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_up(row_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `col_count` times right.
+    pub fn move_right(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_right(col_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `row_count` times down.
+    pub fn move_down(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_down(row_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `col_count` times left.
+    pub fn move_left(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_left(col_count)?;
+        Ok(self)
+    }
+
+    /// Saves the cursor position.
+    ///
+    /// See the [restore_position](struct.TerminalCursor.html#method.restore_position) method.
+    ///
+    /// # Notes
+    ///
+    /// The cursor position is stored globally and is not related to the current / any
+    /// `TerminalCursor` instance.
+    pub fn save_position(&self) -> Result<()> {
+        self.cursor.save_position()
+    }
+
+    /// Restores the saved cursor position.
+    ///
+    /// See the [save_position](struct.TerminalCursor.html#method.save_position) method.
+    pub fn restore_position(&self) -> Result<()> {
+        self.cursor.restore_position()
+    }
+
+    /// Hides the cursor.
+    ///
+    /// See the [show](struct.TerminalCursor.html#method.show) method.
+    pub fn hide(&self) -> Result<()> {
+        self.cursor.hide()
+    }
+
+    /// Shows the cursor.
+    ///
+    /// See the [hide](struct.TerminalCursor.html#method.hide) method.
+    pub fn show(&self) -> Result<()> {
+        self.cursor.show()
+    }
+
+    /// Enables or disables the cursor blinking.
+    ///
+    /// # Notes
+    ///
+    /// Not all terminals do support this functionality. Windows versions lower than
+    /// Windows 10 do not support this functionality.
+    pub fn blink(&self, blink: bool) -> Result<()> {
+        self.cursor.blink(blink)
+    }
+}
+
+/// Creates a new terminal cursor.
```

Comment:
```suggestion
/// Creates a new `Terminal Cursor`.
```

Reply:
Why this one? TerminalCursor says _A terminal cursor._  And a `new` says _Creates a new terminal cursor.` Since we're documenting `TerminalCursor`, why to use backticks here?

Reply:
Because that is returned from this function, I see this function as a constructor/factor function that returns some instance

Reply:
Pushed.